### PR TITLE
[Icons] Add Iconify tests

### PR DESCRIPTION
**Pull request by smnandre**
+ remove Icon implementation

### CODE REVIEW

File: src/Icons/src/Iconify.php [hunk 1]
```diff
@@ -25,10 +25,12 @@
  */
 final class Iconify
 {
+    private const API_BASE_URI = 'https://api.iconify.design';
```
Comment (kbond) [inline src/Icons/src/Iconify.php:28]: Let's use this constant in the bundle config.

**Review by kbond — CHANGES_REQUESTED** [review]
Looks good, just want to note that the iconify API doesn't return 404 in some cases. Just want to ensure this is handled.

File: src/Icons/src/Iconify.php [hunk 2]
```diff
@@ -66,15 +74,15 @@ public function fetchIcon(string $prefix, string $name): Icon
 
     public function fetchSvg(string $prefix, string $name): string
     {
-        $content = $this->http
-            ->request('GET', sprintf('/%s/%s.svg', $prefix, $name))
-            ->getContent()
-        ;
+        $response = $this->http->request('GET', sprintf('/%s/%s.svg', $prefix, $name));
```
Comment (kbond) [inline src/Icons/src/Iconify.php:77]: I'm not sure a non 200 is ever returned. Is this covered by a test? (Same with the other spots we check for a 200)

Comment (smnandre) [reply src/Icons/src/Iconify.php:77]: Yes if the set does not exists / is not available. That's why i added the tests :) 

* https://iconify.design/docs/api/collections.html#error-response
* https://iconify.design/docs/api/collection.html#error-response
* https://iconify.design/docs/api/icon-data.html#error-response

Comment (kbond) [reply src/Icons/src/Iconify.php:77]: Okay, I couldn't remember the exact scenarios were a 404 wasn't returned when I thought it should be.

Comment (smnandre) [reply src/Icons/src/Iconify.php:77]: When the prefix exists but not the icon i think (within the collection endpoint)

**Comment by smnandre** [comment]
Closed and partially merged into #1617

**Pull request closed without merging**